src/components/Table/index.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react"
+import {
+  Box,
+  Table as ChakraTable,
+  TableContainer,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  ThemingProps,
+  Tr,
+} from "@chakra-ui/react"
+
+interface TableProps extends ThemingProps<"Table"> {
+  tableData: { head: string; data: string[] }[]
+}
+
+const Table = (props: TableProps) => {
+  const { tableData, ...rest } = props
+
+  const tableHeadData = tableData.reduce<string[]>((prev, curr) => {
+    return [...prev, curr.head]
+  }, [])
+
+  const tableContentData = tableData
+    .map((item) => item.data)
+    .reduce<string[][]>((prev, curr) => {
+      const splitCurrData = curr.map((item) => item)
+      return [...prev, splitCurrData]
+    }, [])
+
+  const reducedTableContentData = tableContentData.reduce<string[][]>(
+    (_, curr, currIdx, arr) => {
+      let mergeArrs = curr.map((item, idx) => {
+        const prevArr = arr[currIdx - 1]
+
+        return [prevArr ? prevArr[idx] : arr[0][idx], item]
+      })
+      return mergeArrs
+    },
+    []
+  )
```

Comment:
This logic is too constrained, to only consider only two columns of data, when there might be more. Currently redoing.